### PR TITLE
Add ansible_network_direct_execution

### PR DESCRIPTION
**Pull request by cidrblock**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/693

**Need to figure out integration tests for this prior to merge**
**This will require a backport PR for 2.9 in task executor so 2.9 fully qualfied modules use the platform action from the same collection**

This PR add a new connection parameter call `ansible_network_direct_execution`. 

When enabled actions using `plugins/action/network.py ActionModule` as their base class, this include platform actions (cisco.nxos.noxs etc) and  other actions (cli_command), the task's module will be imported by the action
instead of it being tarred to the /tmp/directory.

example:
```
from ansible_collections.ansible.netcommon.plugins.action.network import (
    ActionModule as ActionNetworkModule,
)
```

`cli_command` was updated to the the base class above.

### CODE REVIEW

**Comment by cidrblock** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by cidrblock** [comment]
recheck

**Comment by cidrblock** [comment]
recheck

**Comment by cidrblock** [comment]
recheck

**Comment by cidrblock** [comment]
recheck

File: plugins/action/network.py [hunk 1]
```diff
@@ -205,3 +254,163 @@ def _get_network_os(self, task_vars):
...
 
         return network_os
+
+    def _check_dexec_eligibility(self, host):
+        """ Check if current python and task are eligble
+        """
+        dexec = self.get_connection_option("direct_execution")
+
+        # FIXME: Prior to a merge
+        dexec = True
```
Comment (cidrblock) [inline plugins/action/network.py:264]: Reminder to remove prior to merge

**Comment by Qalthos** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Review by cidrblock — COMMENTED** [review]
looks good

**Review by Qalthos — APPROVED** [review]

**Comment by Qalthos** [comment]
recheck

**Review by ganeshrn — APPROVED** [review]

**Comment by Qalthos** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Pull request merged**